### PR TITLE
Migrate one-off Select inputs to Ant

### PR DESCRIPTION
**Pull request by gilluminate**
Closes [HJ-71](https://ethyca.atlassian.net/browse/HJ-71)

### Description Of Changes

Migrating all of the "one-off" Select components to use the Ant Select component. There is more work to be done on the "custom" Select components, but that will come in another PR.

### Code Changes

* Select migration of select components (see what I did there?)
* Customized Taxonomy options
* Update Cypress tests with different element selectors
* New FilterSelect wrapper component for convenience to keep Filters consistent
* Updated layout of Dataset entry page to be stacked vertically instead of horizontally
* Replace language list from disabled Chakra Select with tags, to just use Ant tags (this was a hack since Chakra didn't have tags)

NOTE: className changes
`.select-*__menu` -> `.ant-select-dropdown`
`.select-*__option` -> `.ant-select-item`

### Steps to Confirm

* Visit http://localhost:3000/consent/privacy-notices
* Click [Add a privacy notice] button
* About 1/2 way down the screen, click [Add notice children +] button
* Button turns into new Select
* This Select should be searchable
* Clicking an option should revert the Select back to a Button

<hr/>

* Add a new system by visiting http://localhost:3000/add-systems/manual
* In the "System details" section, add a few tags to the "System tags" field.
* In the "Data processing properties" section, enable "This system performs profiling" toggle which reveals a Select menu
* In the "Data processing properties" section, enable "This system transfers data" toggle which reveals another Select menu
* Ensure this form still works when the following options have values: "Legal basis for profiling," "Legal basis for transfer," "Responsibility"

<hr/>

* Visit Data map report at http://localhost:3000/reporting/datamap
* click the export button
* Select an option and click download to ensure the correct filetype downloads.

<hr/>

* visit the System Inventory page http://localhost:3000/systems
* Add a new system and configure the Iterable integration using Iterable API Key found in 1Password
* Once added, a new "Consent Automation" accordion button will appear. Click to expand it
* Make some selections from the list of available Privacy Notices and save
* Ensure the form saves as expected

<hr/>

* Configure a Datastore connection (like BigQuery) by going to http://localhost:3000/datastore-connection and then clicking the kebab menu of the connection and clicking "configure"
* Switch to the Dataset configuration tab
* ensure the dataset select works as expected

<hr/>

* visit Connection manager http://localhost:3000/datastore-connection
* Filter by Connection type, System type, Testing status, and Status.
* Ensure each filter behaves as expected
* Use the included (x) clear button to remove any filters

<hr/>

* Visit System Inventory http://localhost:3000/systems
* Click an available system
* Switch to Integrations tab
* Choose "Link Integration" option
* Select an integration to link
* In the resulting form, choose datasets from the new Select

<hr/>

* Visit Messaging http://localhost:3000/messaging
* Click "Add Message +" button
* In the modal, make a selection with the new Select
* Ensure expected behavior

<hr/>

* Visit Manage privacy notices http://localhost:3000/consent/privacy-notices
* Click to edit a privacy notice that has more than one location
* Notice the locations are displayed as pure Tags instead of a disabled Select with the tags represented. (This was originally done because Chakra didn't have any tags and we were piggy backing off the old Select component to solve it)

<hr/>

* Visit Manage privacy notices http://localhost:3000/consent/privacy-notices
* Click "Add a privacy notice"
* At the bottom of the screen click "+ Add a language" button
* Select a language from the Ant select

<hr/>

* Visit Manage datasets (http://localhost:3000/dataset)
* Click rows at least 3 levels deep until you can give rows categories in the Data Categories column
* Click the [+] button to switch over to the Taxonomy Select
* Ensure behavior is as expected

### Pre-Merge Checklist

* [x] Issue requirements met
* [x] All CI pipelines succeeded
* [x] `CHANGELOG.md` updated
* Followup issues:
  * [x] Followup issues created
  * [ ] No followup issues
* Database migrations:
  * [ ] Ensure that your downrev is up to date with the latest revision on `main`
  * [ ] Ensure that your `downgrade()` migration is correct and works
    * [ ] If a downgrade migration is not possible for this change, please call this out in the PR description!
  * [x] No migrations
* Documentation:
  * [ ] Documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] Documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
  * [ ] If there are any new client scopes created as part of the pull request, remember to update public-facing documentation that references our scope registry
  * [x] No documentation updates required

[HJ-71]: https://ethyca.atlassian.net/browse/HJ-71?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jpople — CHANGES_REQUESTED** [review]
Tested locally. Almost all working as intended:

* [x] "Select notice children" on privacy notices
* [x] Datamap report "export" file type
* [X] Consent automation notice selects
* [x] Datastore connection "dataset config"
* [x] Connection manager filters
* [x] System integrations dataset select
* [x] Messaging type
* [x] Privacy notice locations
* [x] Privacy notice language picker
* [x] Taxonomy select

I'm running into some bugs around the DictSuggestionSelects in the system form (tags, legal basis for profiling, legal basis for transfers, responsibility).  

On a clean form, they're populated with a blank tag instead of empty:

Screenshot 2024-11-12 at 17 15 45

And are throwing errors when I try to change the value: 

https://github.com/user-attachments/assets/668d65bc-8646-411f-b3a4-5dc8cdcc7ba5

Also, for "tags", on main we have this "Create [tag name]..." to indicate that users are creating new tags, is it feasible to include something like that on Ant?  With just a blank select menu I think it's a little unclear how to use the field.

Screenshot 2024-11-12 at 17 20 34

**Comment by gilluminate** [comment]
@jpople I think my latest commit solves those issues if you can pull latest. I'll investigate the "creating" question...

**Comment by gilluminate** [comment]
@jpople it doesn't look like that option exists, but I created a feature request to see if they'll implement it for us https://github.com/ant-design/ant-design/issues/51600

**Comment by gilluminate** [comment]
@jpople ok, I actually figured out how to get it working with the current options. Not as elegant as a new prop, but it works :)

File: clients/admin-ui/src/features/system/dictionary-form/DictSuggestionInputs.tsx [hunk 1]
```diff
@@ -310,197 +291,32 @@ export const DictSuggestionSelect = ({
...
-            isDisabled={disabled}
-            isMulti={isMulti}
+            optionRender={
+              mode === "tags"
+                ? (option) => {
+                    if (
+                      option.value === searchValue &&
+                      !field.value.includes(searchValue)
+                    ) {
+                      return `Create "${searchValue}"`;
```
Comment (jpople) [inline clients/admin-ui/src/features/system/dictionary-form/DictSuggestionInputs.tsx:309]: Nice workaround here!

**Review by jpople — APPROVED** [review]
Thanks for the changes-- looking good!

**Review by Roger-Ethyca — APPROVED** [review]
there will be some changes for E2E tests.

**Pull request merged**